browser/ui/webui/ai_chat/ai_chat_ui.h

```diff
@@ -50,12 +55,22 @@ class AIChatUI : public ui::UntrustedWebUIController {
   WEB_UI_CONTROLLER_TYPE_DECL();
 };
 
+#if !BUILDFLAG(IS_ANDROID)
 class UntrustedChatUIConfig : public DefaultTopChromeWebUIConfig<AIChatUI> {
```

Comment:
why are we even using `DefaultTopChromeWebUIConfig` if we don't actually need it?

Reply:
We do need it on Desktop. It's needed for auto-resizing call that is used in WebUIContentsWrapperT (https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/top_chrome/webui_contents_wrapper.h;l=185)

Reply:
do we really need this buildflag mess just to return false from `ShouldAutoResizeHost` ?

Reply:
Would it be more palatable is we used a slightly different syntax like:
```
#if !BUILDFLAG(IS_ANDROID)
using AIChatUIConfigBase = DefaultTopChromeWebUIConfig<AIChatUI>;
#else
using AIChatUIConfigBase = content::WebUIConfig;
#endif  // #if !BUILDFLAG(IS_ANDROID)

class UntrustedChatUIConfig : public AIChatUIConfigBase {
```

Reply:
It's not the end of the world, but if all we care about is `ShouldAutoResizeHost` it seems simpler to just override that directly